engine/src/button.cpp

```diff
@@ -412,8 +410,11 @@ MCButton::~MCButton()
 
 void MCButton::removelink(MCObject *optr)
 {
-	if (optr == menu)
-		menu = NULL;
+	if (menu.IsValid() && optr == menu)
```

Comment:
This should probably be "IsBound()" here - we want to check if the handle currently points at something, not whether that something still exists or not.